src/fields/vec_seq_double.hh

```diff
@@ -237,8 +260,111 @@ private:
     VectorDataPtr data_ptr_;
     /// stored vector of data in PETSC format
     Vec data_petsc_;
+    /// communicator
+    MPI_Comm communicator_;
 };
 
 
+/**
+ * Helper method creating VectorMPI of given size with serial Petsc communicator.
+ *
+ * Method is used for better readability of code.
+ */
+inline VectorMPI * create_vector_mpi(unsigned int size)
```

Comment:
Make a static method of VectorMPI, rename to sequential. Pass PETSC_COMM_SELF explicitely.